Rewrite Promise constructor description (#17406)
mdn/content: files/en-us/web/javascript/reference/global_objects/promise/promise/index.md

### Parameters

- `executor`

>   - : A {{jsxref("function")}} to be executed by the constructor. It receives two functions as parameters: `resolutionFunc` and `rejectionFunc`. Any errors thrown in the `executor` will cause the promise to reject, and the return value is neglected. The semantics of `executor` are explained in detail below. 

Reviewer: ```suggestion
  - : A {{jsxref("function")}} to be executed by the constructor. It receives two functions as parameters: `resolutionFunc` and `rejectionFunc`. Any errors thrown in the `executor` will cause the promise to be rejected, and the return value will be neglected. The semantics of `executor` are detailed below
```

### Return value

> When called via `new`, the `Promise` constructor returns a promise object. The promise object will become "resolved" when either of the functions `resolutionFunc` or `rejectionFunc` are invoked. Note that if you call `resolutionFunc` or `rejectionFunc` and pass another Promise object as an argument, it can be said to be "resolved", but still not "settled".

Author: There's no place on top of my mind (but I don't typically read those guide pages unless someone reports an issue😬). The `Promise.resolve()` page is also very unclear on this (and someone has already complained in https://github.com/mdn/content/issues/334) so I'll probably eventually write one somewhere.

## Description

> Traditionally (before promises), asynchronous tasks are designed as callbacks.

Reviewer: ```suggestion
Traditionally (before promises), asynchronous tasks were designed as callbacks.
```

  // final `error` or `result`. Any operation dependent on the
  // result must be defined within this callback.
});
// Code here is immediately executed after the `readFile` request
// is fired. It does not wait for the callback to be called, hence
// making `readFile` "asynchronous".
```

To take advantage of the readability improvement and language features offered by promises, the `Promise()` constructor allows one to transform the callback-based API to a promise-based one.

> **Note:** If your task is already promise-based, you likely do not need the `Promise()` constructor.

The `executor` is custom code that ties an outcome in a callback to a promise. You, the programmer, write the `executor`. Its signature is expected to be:

```js
function (resolutionFunc, rejectionFunc) {
  // Typically, some asynchronous operation that accepts a callback,
  // like the `readFile` function above
}
```

`resolutionFunc` and `rejectionFunc` are also functions, and you can give them whatever actual names you want. Their signatures are simple: they accept a single parameter of any type.

```js
resolutionFunc(value) // call on resolved
rejectionFunc(reason) // call on rejected
```

> The `resolutionFunc` `value` parameter can be another promise object, in which case the promise gets dynamically inserted into the [promise chain](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#chained_promises). The `rejectionFunc` has semantics close to the [`throw`](en-US/docs/Web/JavaScript/Reference/Statements/throw) statement, so `reason` is typically an [`Error`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) instance. If `value` or `reason` is omitted, the promise is resolved/rejected with `undefined`.

Reviewer: ```suggestion
The `resolutionFunc` `value` parameter can be another promise object, in which case the promise gets dynamically inserted into the [promise chain](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#chained_promises). The `rejectionFunc` has semantics close to the [`throw`](en-US/docs/Web/JavaScript/Reference/Statements/throw) statement, so `reason` is typically an [`Error`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) instance. If either the `value` or the `reason` are omitted, the promise is resolved/rejected with `undefined.`
```

Author: either...or is used as singular.
